packages/atomic/src/components/commerce/atomic-product-description/atomic-product-description.spec.ts

```diff
@@ -0,0 +1,136 @@
+import {html} from 'lit';
+import {describe, expect, it} from 'vitest';
+import {renderInAtomicProduct} from '@/vitest-utils/testing-helpers/fixtures/atomic/commerce/atomic-product-fixture';
+import {buildFakeProduct} from '@/vitest-utils/testing-helpers/fixtures/headless/commerce/product';
+import '../atomic-product-text/atomic-product-text';
+import {userEvent} from '@vitest/browser/context';
+import {ifDefined} from 'lit/directives/if-defined.js';
+import type {TruncateAfter} from '../../common/expandable-text/expandable-text';
+import './atomic-product-description';
+
+describe('atomic-product-description', () => {
+  const renderProductDescription = async (
+    props: {
+      truncateAfter?: TruncateAfter;
+      isCollapsible?: boolean;
+      field?: 'ec_description' | 'ec_shortdesc';
+      ec_shortdesc?: string;
+      ec_description?: string;
+    } = {}
+  ) => {
+    const mockedProduct = buildFakeProduct({
+      ec_description:
+        props.ec_description ??
+        `This is a description for child product. It contains detailed information about the product features and specifications. It should be informative and helpful for potential buyers.`,
+      ec_shortdesc:
+        props.ec_shortdesc ??
+        `This is a short description for child product. It is meant to be concise and informative. It should give a brief overview of the product features and benefits.`,
+    });
+
+    const {element} = await renderInAtomicProduct({
+      template: html`
+      <atomic-product-description
+        truncate-after=${ifDefined(props.truncateAfter)}
+        ?is-collapsible=${props.isCollapsible ?? false}
+        field=${ifDefined(props.field)}
+      ></atomic-product-description>
+      `,
+      selector: 'atomic-product-description',
+      product: mockedProduct,
+    });
+
+    return {
+      element,
+      locators: {
+        get text() {
+          return element.querySelector('atomic-commerce-text');
+        },
+        get expandableDiv() {
+          return element.querySelector('.expandable-text');
+        },
+        get button() {
+          return element.querySelector('button');
+        },
+      },
+    };
+  };
+
```

Comment:
We should test that it validates the props upon update (i.e., expect error when you pass an invalid prop)

Also, I think we should test that we disconnect the resize observer when the component is removed from the DOM (e.g., by spying on the ResizeObserver class' `disconnect` method directly).